modules/finance/src/main/java/com/opengamma/strata/finance/rate/bond/ExpandedFixedCouponBond.java

```diff
@@ -0,0 +1,736 @@
+
+import org.joda.beans.Bean;
+import org.joda.beans.BeanDefinition;
+import org.joda.beans.ImmutableBean;
+import org.joda.beans.JodaBeanUtils;
+import org.joda.beans.MetaProperty;
+import org.joda.beans.Property;
+import org.joda.beans.PropertyDefinition;
+import org.joda.beans.impl.direct.DirectFieldsBeanBuilder;
+import org.joda.beans.impl.direct.DirectMetaBean;
+import org.joda.beans.impl.direct.DirectMetaProperty;
+import org.joda.beans.impl.direct.DirectMetaPropertyMap;
+
+import com.google.common.collect.ImmutableList;
+import com.opengamma.strata.basics.currency.Currency;
+import com.opengamma.strata.basics.currency.Payment;
+import com.opengamma.strata.basics.date.DayCount;
+import com.opengamma.strata.basics.date.DaysAdjustment;
+import com.opengamma.strata.collect.id.StandardId;
+
+/**
+ * An expanded fixed coupon bond.
+ * <p>
+ * A fixed coupon bond is a financial instrument that represents a stream of fixed payments. 
+ * The payments consist two types: periodic coupon payments and nominal payment.
+ * The periodic payments are made {@code n} times a year with a fixed coupon rate at individual coupon dates.   
+ * The nominal payment is the unique payment at the final coupon date.
+ * <p>
+ * The list of {@link FixedCouponBondPaymentPeriod} represents the periodic coupon payments,  
+ * whereas the nominal payment is defined by {@link Payment}.
+ * <p>
+ * The accrual factor between two dates is computed {@code dayCount}. 
+ * The legal entity of this fixed coupon bond is identified by {@link StandardId}. The enum, {@link YieldConvention}, 
+ * specifies the yield computation convention, and {@link DaysAdjustment} does the number of days between valuation 
+ * date and settlement date. 
+ * </ul>
+ */
+@BeanDefinition
+public final class ExpandedFixedCouponBond
+    implements FixedCouponBondProduct, ImmutableBean, Serializable {
+
+  /**
+   * The nominal payment of the product.
+   * <p>
+   * The payment date of the nominal payment agrees with the final coupon payment date of the periodic payments.
+   */
+  @PropertyDefinition(validate = "notNull")
+  private final Payment nominalPayment;
+  /**
+   * The periodic payments of the product.
+   * <p>
+   * Each payment period represents part of the life-time of the product.
+   * The start date and end date of the leg are determined from the first and last period.
+   * As such, the periods should be sorted.
+   */
+  @PropertyDefinition(validate = "notNull")
+  private final ImmutableList<FixedCouponBondPaymentPeriod> periodicPayments;
+  /**
+   * The day count convention applicable. 
+   * <p>
+   * The conversion from dates to a numerical value is made based on this day count. 
+   * For the fixed bond, the day count convention is used to compute accrued interest.
+   */
+  @PropertyDefinition(validate = "notNull")
+  private final DayCount dayCount;
+  /**
+   * Yield convention.
+   * <p>
+   * The convention defines accrued interest calculation type of the product.  
```

Comment:
replace by something like: The convention defines how to convert from yield to price and inversely."